src/Tree/HydroTree.cpp

```diff
@@ -1922,13 +1922,15 @@ void HydroTree<ndim,ParticleType>::CheckValidNeighbourList
     for (j=0; j<Ntot; j++) {
       for (k=0; k<ndim; k++) dr[k] = partdata[j].r[k] - partdata[i].r[k];
       drsqd = DotProduct(dr,dr,ndim);
+      if (partdata[j].flags.is_dead()) continue;
```

Comment:
Can you add these checks to the version of this function in the neighbour manager?